README.md

```diff
@@ -479,6 +509,89 @@ registry.define('my-widget', () => {
 });
 ```
 
+#### Injecting State
+
+Working with larger widget structures, it can quickly become tiresome and complex to pass all the required properties down the tree. Additionally having to do so often means that widgets are forced to be aware of properties that they they don't need, but are required to know about in order to ensure they are propagated to the children.
```

Comment:
"Additionally having to do so often means that widgets are forced to be aware of properties that they they don't need, but are required to know about in order to ensure they are propagated to the children." is a bit hard to follow, perhaps something like "Needing to frequently pass all required properties also requires widgets to be aware of properties that are only needed for ensuring their propagation to child widgets."